### PR TITLE
Add bulk deletion for images

### LINKED ISSUE #1912: Images - Bulk Deletion
Add the ability to select multiple images for deletion.

2-images

### PR DESCRIPTION
**Pull request by rak-phillip**
This adds a bulk deletion option to images that have the delete option available by making use of the `bulkable` and `bulkActions` properties for `availableActions`.

closes #1912 

#### Images with a Delete Action, None Selected

Screen Shot 2022-05-12 at 4 34 33 PM

#### Images with a Delete Action, All Selected

Screen Shot 2022-05-12 at 4 34 45 PM

#### Images with a Delete Action, One Selected

Screen Shot 2022-05-12 at 4 34 55 PM

#### Delete Action Available for a Subset of Images

Screen Shot 2022-05-12 at 4 35 17 PM

### CODE REVIEW

**Comment by ericpromislow** [comment]
I'm getting these eslint messages when I run `npm t` on macos -- not sure why, as not all of them are in the changed code, unless a bump in something like `eslint` is causing these. 

```
/Users/ericp/workspace/rancher/desktop/src/components/ActionDropdown.vue
  71:37  warning  Attribute ':buttonSize' must be hyphenated  vue/attribute-hyphenation

/Users/ericp/workspace/rancher/desktop/src/components/ImagesOutputWindow.vue
  138:9  warning  Attribute ':isLoading' must be hyphenated  vue/attribute-hyphenation
  150:9  warning  Attribute ':hasError' must be hyphenated   vue/attribute-hyphenation
  162:9  warning  Attribute ':isSuccess' must be hyphenated  vue/attribute-hyphenation

/Users/ericp/workspace/rancher/desktop/src/components/SortableTable/index.vue
  851:62  warning  Attribute ':fullColspan' must be hyphenated  vue/attribute-hyphenation
  890:21  warning  Attribute ':rowKey' must be hyphenated       vue/attribute-hyphenation

/Users/ericp/workspace/rancher/desktop/src/components/form/RadioGroup.vue
  159:11  warning  Attribute ':isDisabled' must be hyphenated  vue/attribute-hyphenation

✖ 7 problems (0 errors, 7 warnings)
  0 errors and 7 warnings potentially fixable with the `--fix` option.
```

**Comment by ericpromislow** [comment]
> This adds a bulk deletion option to images that have the delete option available by making use of the bulkable and bulkActions properties for availableActions.

Noting that the `availableActions` object has neither of these named properties. But I see similar names, and that we're inheriting these from `dashboard`, so I can live with the discrepancy.

File: src/components/Images.vue [hunk 1]
```diff
@@ -250,6 +271,42 @@ export default {
     startRunningCommand(command) {
       this.imageOutputCuller = getImageOutputCuller(command);
     },
+    async deleteImages() {
+      const message = `Delete ${ this.imagesToDelete.length } ${ this.imagesToDelete.length > 1 ? 'images' : 'image' }?`;
+      const detail = this.imagesToDelete.reduce((prev, curr) => {
+        if (!this.isDeletable(curr)) {
```
Comment (ericpromislow) [inline src/components/Images.vue:277]: From my reading, `this.isDeletable(curr)` should always be `true`.

But I would replace the `reduce` with the simpler

```
this.imagesToDelete.map(image => `${ image.imageName }:${ image.tag }`).join("\n")
```

File: src/k8s-engine/images/imageProcessor.ts [hunk 2]
```diff
@@ -379,6 +379,8 @@ export abstract class ImageProcessor extends EventEmitter {
 
   abstract deleteImage(imageID: string): Promise<childResultType>;
 
+  abstract deleteImages(images: string[]): Promise<childResultType>;
```
Comment (ericpromislow) [inline src/k8s-engine/images/imageProcessor.ts:382]: These look more like `imagesIDs` to me than `images` (which should usually refer to the image object, not a string).

File: src/k8s-engine/images/mobyImageProcessor.ts [hunk 3]
```diff
@@ -46,6 +46,10 @@ export default class MobyImageProcessor extends imageProcessor.ImageProcessor {
     return await this.runImagesCommand(['rmi', imageID]);
   }
 
+  async deleteImages(images: string[]): Promise<imageProcessor.childResultType> {
```
Comment (ericpromislow) [inline src/k8s-engine/images/mobyImageProcessor.ts:49]: See above comment on `images` vs `imageIDs`

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Works fine, just a few things to clear up the code a bit.

File: src/components/Images.vue (hunk 1 above)
Comment (rak-phillip) [reply src/components/Images.vue]: That's a much more elegant refactor, thanks and good catch 😁

I think my original implementation required the `reduce`, but that is obviously no longer needed here.

**Review by ericpromislow — COMMENTED** [review]
So the code's fine, but I just noticed that when I delete more than one image, only the details for the final deletion show up in the `Image Output` textarea. Do we care about this? I see 3 options here:
1. Current behavior in the PR (just show output for the final deletion)
2. Collect output for all deletions and display everything in the textarea.
3. Just show the summary for all deletions.

I prefer option 2. Leaving as a 'Comment' because I might be wrong, option 1 is fine, and then this is ready for approval.

**Review by ericpromislow — APPROVED** [review]
This is fine for now. I'll add an issue for the output issue.

**Pull request merged**